File: app/src/main/java/com/example/mobile/julfani/tubes/utils/SleepListAdapter.java
package com.example.mobile.julfani.tubes.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mobile.julfani.tubes.R;
import com.example.mobile.julfani.tubes.entity.Sleep;

import java.util.List;

class SleepViewHolder extends RecyclerView.ViewHolder {
    public final TextView textView;

    public SleepViewHolder(@NonNull View itemView) {
        super(itemView);
        textView = itemView.findViewById(R.id.recylceview_textview);
    }
}

public class SleepListAdapter extends RecyclerView.Adapter<SleepViewHolder> {

    private List<Sleep> sleepList;
    private final LayoutInflater inflater;

    public SleepListAdapter(Context context) {
        inflater = LayoutInflater.from(context);
    }


    @NonNull
    @Override
    public SleepViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = inflater.inflate(R.layout.recycleview, parent, false);
        return new SleepViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull SleepViewHolder holder, int position) {
        String res = sleepList.get(position).getId() + " " + Utils.formatTime(sleepList.get(position).getDuration());

        holder.textView.setText(res);
    }

    @Override
    public int getItemCount() {
        if(sleepList != null) return sleepList.size();
        return 0;
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setSleepList(List<Sleep> sleepList) {
        this.sleepList = sleepList;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/nursyah/bettersleep/utils/SleepListAdapter.kt
package com.nursyah.bettersleep.utils

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.nursyah.bettersleep.R
import com.nursyah.bettersleep.entity.Sleep

class SleepViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val textView: TextView
    init { textView = itemView.findViewById(R.id.textview_sleep_recycle) }
}

class SleepListAdapter(context: Context?) : RecyclerView.Adapter<SleepViewHolder>() {
    private var sleepList: List<Sleep>? = null
    private val inflater: LayoutInflater

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SleepViewHolder {
        val itemView = inflater.inflate(R.layout.recycleview_sleep, parent, false)
        return SleepViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: SleepViewHolder, position: Int) {
        val res = "${sleepList!![position].date} ${Utils.formatTime(sleepList!![position].duration)}"
        holder.textView.text = res
    }


    override fun getItemCount(): Int {
        return if (sleepList != null) sleepList!!.size else 0
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setSleepList(sleepList: List<Sleep>?) {
        this.sleepList = sleepList
        notifyDataSetChanged()
    }

    init { inflater = LayoutInflater.from(context) }
}